### PR TITLE
ping db in health check

### PR DESCRIPTION
**Pull request by iSchluff**
Add basic db check to health endpoint

- [x] read the [CONTRIBUTING guidelines](README.md#contributing)
- [ ] raised a GitHub issue or discussed it on the projects chat beforehand
- [ ] added unit tests
- [ ] added integration tests
- [ ] updated documentation if needed
- [ ] updated CHANGELOG.md

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -30,6 +30,42 @@ const (
...
+		buf, err := json.Marshal(res)
+		if err != nil {
+			log.Error().Caller().Err(err).Msg("marshal failed")
+		}
+		writer.Write(buf)
+	}
+
+	err := h.pingDB()
+	if err != nil {
+		respond(err)
```
Comment (kradalby) [inline api.go:62]: We should not respond the error to the user, it is consider bad practice by OWASP;

https://cheatsheetseries.owasp.org/cheatsheets/Error_Handling_Cheat_Sheet.html

File: api.go [hunk 2]
```diff
@@ -30,6 +30,42 @@ const (
...
 
+func (h *Headscale) HealthHandler(
+	writer http.ResponseWriter,
+	req *http.Request,
+) {
+	respond := func(err error) {
+		writer.Header().Set("Content-Type", "application/json; charset=utf-8")
+
+		res := struct {
+			Healthy string `json:"healthy"`
```
Comment (kradalby) [inline api.go:41]: I am happy with this, just wondering, is this a "defined" standard format for health?

I want:
1. Something is easily consumed by standard health checks, like Docker's, SystemD and Kubernetes
2. The closest to a standard as possible

**Review by kradalby — COMMENTED** [review]
Overall quite happy with this, seems reasonable, posted a couple of comments and would you mind fixing the lint errors?

File: api.go (hunk 1 above)
Comment (iSchluff) [reply api.go:62]: ok will change that

**Review by juanfont — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**